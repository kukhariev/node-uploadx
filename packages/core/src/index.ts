export * from './handlers';
export * from './storages';
export * from './utils';
export * from './types';
