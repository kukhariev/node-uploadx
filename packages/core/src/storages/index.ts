export * from './disk-storage';
export * from './file';
export * from './storage';
export * from './local-meta-storage';
export * from './meta-storage';
export * from './config';
