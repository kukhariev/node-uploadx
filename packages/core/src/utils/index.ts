export * from './cache';
export * from './errors';
export * from './fs';
export * from './http';
export * from './locker';
export * from './logger';
export * from './pipes';
export * from './primitives';
export * from './validator';
