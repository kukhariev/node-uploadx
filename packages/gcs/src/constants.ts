export const BUCKET_NAME = 'node-uploadx';
export const uploadAPI = `https://storage.googleapis.com/upload/storage/v1/b`;
export const storageAPI = `https://storage.googleapis.com/storage/v1/b`;
export const authScopes = ['https://www.googleapis.com/auth/devstorage.full_control'];
