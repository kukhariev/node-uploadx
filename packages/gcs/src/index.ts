export * from './gcs-storage';
export * from './gcs-meta-storage';
export * from './constants';
