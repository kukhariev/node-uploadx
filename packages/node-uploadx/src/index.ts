export * from '@uploadx/core';
export * from '@uploadx/s3';
export * from '@uploadx/gcs';
