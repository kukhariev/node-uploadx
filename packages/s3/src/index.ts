export * from './s3-storage';
export * from './s3-meta-storage';
export * from './aws-error';
