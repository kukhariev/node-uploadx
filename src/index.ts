export * from './core';
export * from './core/utils';
export * from './Tus';
export * from './Uploadx';
export * from './Multipart';
