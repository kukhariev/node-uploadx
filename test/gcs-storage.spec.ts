import { expect } from 'chai';
import { GCStorage } from '../src';

describe('GCStorage', function() {
  before(function() {
    (process.env.CI || !process.env.GCS_BUCKET) && this.skip();
  });

  describe('constructor', () => {
    it('should create gcs-storage', function() {
      const gcs = new GCStorage();
      expect(gcs).to.be.instanceOf(GCStorage);
    });
  });
});
