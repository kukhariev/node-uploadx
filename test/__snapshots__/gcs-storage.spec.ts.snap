// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GCStorage .create() should handle existing 1`] = `
Object {
  "bytesWritten": 6,
  "contentType": "video/mp4",
  "createdAt": "2022-02-02T00:00:00.000Z",
  "id": "11f967df-da1013ca-19e8e887-b9682398",
  "metadata": Object {
    "custom": "",
    "lastModified": 1635398061454,
    "mimeType": "video/mp4",
    "name": "testfile.mp4",
    "sha1": "ZAPAntzKARqtb+j3B529GAOf3kI=",
    "size": 64,
  },
  "name": "userId/testfile.mp4",
  "originalName": "testfile.mp4",
  "size": 64,
  "uri": "http://api.com?upload_id=123456789",
  "userId": "userId",
}
`;

exports[`GCStorage .create() should request api and set status and uri 1`] = `
GCSFile {
  "bytesWritten": NaN,
  "contentType": "video/mp4",
  "createdAt": "2022-02-02T00:00:00.000Z",
  "expiredAt": "2022-02-02T01:00:00.000Z",
  "id": "11f967df-da1013ca-19e8e887-b9682398",
  "metadata": Object {
    "custom": "",
    "lastModified": 1635398061454,
    "mimeType": "video/mp4",
    "name": "testfile.mp4",
    "sha1": "ZAPAntzKARqtb+j3B529GAOf3kI=",
    "size": 64,
  },
  "name": "userId/testfile.mp4",
  "originalName": "testfile.mp4",
  "size": 64,
  "status": "created",
  "uri": "http://api.com?upload_id=123456789",
  "userId": "userId",
}
`;

exports[`GCStorage .create() should request api and set status and uri 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "url": "https://storage.googleapis.com/storage/v1/b/test-bucket",
      },
    ],
    Array [
      Object {
        "params": Object {
          "alt": "media",
        },
        "url": "https://storage.googleapis.com/storage/v1/b/test-bucket/o/11f967df-da1013ca-19e8e887-b9682398.META",
      },
    ],
    Array [
      Object {
        "body": "{\\"metadata\\":{\\"name\\":\\"testfile.mp4\\",\\"size\\":64,\\"mimeType\\":\\"video/mp4\\",\\"lastModified\\":1635398061454,\\"custom\\":\\"\\",\\"sha1\\":\\"ZAPAntzKARqtb+j3B529GAOf3kI=\\"}}",
        "headers": Object {
          "Content-Type": "application/json; charset=utf-8",
          "Origin": "http://api.com",
          "X-Upload-Content-Length": "64",
          "X-Upload-Content-Type": "video/mp4",
        },
        "method": "POST",
        "params": Object {
          "name": "userId/testfile.mp4",
          "size": 64,
          "uploadType": "resumable",
        },
        "url": "https://storage.googleapis.com/upload/storage/v1/b/test-bucket/o",
      },
    ],
    Array [
      Object {
        "body": "{\\"bytesWritten\\":null,\\"name\\":\\"userId/testfile.mp4\\",\\"metadata\\":{\\"name\\":\\"testfile.mp4\\",\\"size\\":64,\\"mimeType\\":\\"video/mp4\\",\\"lastModified\\":1635398061454,\\"custom\\":\\"\\",\\"sha1\\":\\"ZAPAntzKARqtb+j3B529GAOf3kI=\\"},\\"originalName\\":\\"testfile.mp4\\",\\"contentType\\":\\"video/mp4\\",\\"size\\":64,\\"userId\\":\\"userId\\",\\"id\\":\\"11f967df-da1013ca-19e8e887-b9682398\\",\\"uri\\":\\"http://api.com?upload_id=123456789\\",\\"createdAt\\":\\"2022-02-02T00:00:00.000Z\\",\\"expiredAt\\":\\"2022-02-02T01:00:00.000Z\\"}",
        "headers": Object {
          "Content-Type": "application/json; charset=utf-8",
        },
        "method": "POST",
        "params": Object {
          "name": "11f967df-da1013ca-19e8e887-b9682398.META",
          "uploadType": "media",
        },
        "url": "https://storage.googleapis.com/upload/storage/v1/b/test-bucket/o",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`GCStorage .list() should return all user files 1`] = `
Array [
  Object {
    "createdAt": Date { NaN },
    "id": "11f967df-da1013ca-19e8e887-b9682398",
  },
]
`;

exports[`GCStorage .write() should request api and set status and bytesWritten 1`] = `
Object {
  "bytesWritten": 64,
  "contentType": "video/mp4",
  "createdAt": "2022-02-02T00:00:00.000Z",
  "id": "11f967df-da1013ca-19e8e887-b9682398",
  "metadata": Object {
    "custom": "",
    "lastModified": 1635398061454,
    "mimeType": "video/mp4",
    "name": "testfile.mp4",
    "sha1": "ZAPAntzKARqtb+j3B529GAOf3kI=",
    "size": 64,
  },
  "name": "userId/testfile.mp4",
  "originalName": "testfile.mp4",
  "size": 64,
  "status": "completed",
  "uri": "https://storage.googleapis.com/storage/v1/b/test-bucket/o/userId/testfile.mp4",
  "userId": "userId",
}
`;

exports[`GCStorage .write() should request api and set status and bytesWritten on resume 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "http://api.com?upload_id=123456789",
      Object {
        "headers": Object {
          "Accept": "application/json",
          "Content-Range": "bytes */64",
        },
        "method": "PUT",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;
